### PR TITLE
Feature/generic as fuck boiiiii

### PR DESCRIPTION
**Pull request by ghost**
Oh boi this makes one generic as fuck controller.
Tested once, push to master!

### CODE REVIEW

File: src/dndCharApi/Controllers/GenericCharacterController.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            var collection = MongoDb.GetCollection<dynamic>("RpgCharModels");
+            var stringId = id.ToString();
+            var filter = Builders<dynamic>.Filter.Eq("_id", ObjectId.Parse(stringId));
+            var list = await (await collection.FindAsync(filter)).ToListAsync();
+
+            if (list.Count > 0)
+            {
+                var listOfProps = new Dictionary<string, string>();
+                var theOne = list[0] as BaseCharacterSheet;
+                var baseProps = typeof(BaseCharacterSheet).GetProperties(BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Public);
```
Comment (NullableInt) [inline src/dndCharApi/Controllers/GenericCharacterController.cs]: Remove unused variable, bitch

File: src/dndCharApi/Controllers/GenericCharacterController.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                return BadRequest();
+            }   
+
+            var deserializedGenericGoo = Newtonsoft.Json.JsonConvert.DeserializeObject(body.ToString(), type.GetType());
+            if (deserializedGenericGoo == null)
+            {
+                return BadRequest();
+            }
+            
+            var firmlyShapedGoo = deserializedGenericGoo as BaseCharacterSheet;
```
Comment (jaskeland) [inline src/dndCharApi/Controllers/GenericCharacterController.cs:148]: What even are these variablenames.

File: src/dndCharApi/Controllers/GenericCharacterController.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                    {
+                        var value = Convert.ChangeType(propertyValue, thePropertyIfFound.PropertyType);
+                        thePropertyIfFound.SetValue(theOne, value);
+
+                        await collection.UpdateOneAsync(filter, Builders<dynamic>.Update.Set(property, value).CurrentDate("_lastUpdated"));
+                        return Ok(propertyValue);
+                    }
+                    catch (Exception)
+                    {
+                        return BadRequest("fuck");
```
Comment (NullableInt) [inline src/dndCharApi/Controllers/GenericCharacterController.cs]: Fuckin 'ell mate, this aint a good 'n proper message to send to the frontu endu, givem something more juicy

File: src/dndCharApi/Controllers/GenericCharacterController.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+
+        [HttpPost("create/{gameSystem}")]
+        public async Task<IActionResult> CreateSomeCharacterIfYouCanBoi([FromRoute] string gameSystem, [FromBody] dynamic body)
+        {
+            var userId = User.FindFirst(ClaimTypes.NameIdentifier).Value;
+            var type = characterSheetTypes.FirstOrDefault(s => s.GetType().Name.Equals(gameSystem, StringComparison.OrdinalIgnoreCase));
+            if (type == null)
+            {
+                return BadRequest();
```
Comment (NullableInt) [inline src/dndCharApi/Controllers/GenericCharacterController.cs]: Add some message here boiys

File: src/dndCharApi/Controllers/GenericCharacterController.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            }   
+
+            var deserializedGenericGoo = Newtonsoft.Json.JsonConvert.DeserializeObject(body.ToString(), type.GetType());
+            if (deserializedGenericGoo == null)
+            {
+                return BadRequest();
```
Comment (NullableInt) [inline src/dndCharApi/Controllers/GenericCharacterController.cs]: Poor goo 😢 Tell em why it failed 😭

**Review by NullableInt — CHANGES_REQUESTED** [review]
God damn it boy! You can do better than this!

**Review by jaskeland — APPROVED** [review]
Totally unmaintainable, not even tested. Wtf.

Approved.

File: src/dndCharApi/Controllers/GenericCharacterController.cs (hunk 2 above)
Comment (NullableInt) [reply src/dndCharApi/Controllers/GenericCharacterController.cs:148]: They're good names, Brent

**Comment by NullableInt** [comment]
> Totally unmaintainable, not even tested. Wtf.
> 
> Approved.

I'll have you know I tested it at least twice in Postman ™️

**Pull request merged**